### PR TITLE
Pubs

### PR DESCRIPTION
**Pull request by bradfordcondon**
implement publication support for BioProjects #141 

Missing features:

* No formatter for pub.  I dont think we're supporting direct import. (maybe we should?)
* No formatting of publication on the project.
  - The XML just has the pmid.  So, we could get it, but would have to make the extra request.
  - now that im typing it... why not?
  - because it'll break users without an API for starters.  Nevermind.  we'll just format as a list of IDs as with other records for now.

### CODE REVIEW

**Comment by bradfordcondon** [comment]
I realized that adding a full blown formatter is too much, because we dont want to be making all those requests to NCBI.  Obviously we could make a single request with all IDs, but unfortunately i dont think the core API is hooked up to work that way (i might be wrong about that).

So anyway for now we just have an ugly table of links:

screen shot 2019-02-08 at 4 53 45 pm

I think it's feasible we alter this to be titles instead and fetch the records.

**Comment by bradfordcondon** [comment]
shoot- another case where the test passes in isolation but fails in the suite.

Interestingly, I can get it to pass by changing the `$title` variable (two pubs are inserted).  Seems like a good place to start troubleshooting.

**Comment by bradfordcondon** [comment]
function test confirmation:

```

2019-02-08 22:55:25: Calling: tripal_eutils_create_records(bioproject, 12384, 1)
INFO (TRIPAL_EUTILS): Inserting record into Chado: bioproject: 12384
INFO (TRIPAL_EUTILS): Inserting record into Chado: pubmed: 9878232
INFO (TRIPAL_EUTILS): Inserting record into Chado: pubmed: 16341006
INFO (TRIPAL_EUTILS): Inserting record into Chado: pubmed: 9023104
INFO (TRIPAL_EUTILS): Inserting record into Chado: pubmed: 1315422
INFO (TRIPAL_EUTILS): Inserting record into Chado: pubmed: 10687853
INFO (TRIPAL_EUTILS): Inserting record into Chado: pubmed: 10806195
INFO (TRIPAL_EUTILS): Inserting record into Chado: pubmed: 16647053
INFO (TRIPAL_EUTILS): Inserting record into Chado: pubmed: 17560599
INFO (TRIPAL_EUTILS): Inserting record into Chado: pubmed: 20806078
INFO (TRIPAL_EUTILS): Inserting record into Chado: pubmed: 21643798
INFO (TRIPAL_EUTILS): Inserting record into Chado: pubmed: 22281395
INFO (TRIPAL_EUTILS): Inserting record into Chado: pubmed: 22342001
INFO (TRIPAL_EUTILS): Inserting record into Chado: pubmed: 22623710
INFO (TRIPAL_EUTILS): Inserting record into Chado: pubmed: 22988997
INFO (TRIPAL_EUTILS): Inserting record into Chado: pubmed: 24026177
INFO (TRIPAL_EUTILS): Inserting record into Chado: pubmed: 8698819
INFO (TRIPAL_EUTILS): Inserting record into Chado: pubmed: 10964405
INFO (TRIPAL_EUTILS): Inserting record into Chado: pubmed: 24034269
```

**Comment by bradfordcondon** [comment]
@almasaeed2010 i've added a new repository and parser class.  These classes just wrap core API but still if you've got the time (like, next week) to glance at this PR and make sure i'm following our conventions properly I'd appreciate it!

File: includes/xml_parsers/EUtilsPubmedParser.inc [hunk 1]
```diff
@@ -0,0 +1,32 @@
+<?php
+
+/**
+ * Class EUtilsPubmedParser.
+ *
+ * Note that projects don't have reliable attribute listings.
```
Comment (bradfordcondon) [inline includes/xml_parsers/EUtilsPubmedParser.inc]:
```suggestion
```

**Comment by almasaeed2010** [comment]
Will do!

**Comment by almasaeed2010** [comment]
The structure looks correct to me 👍

**Pull request merged**